vulture_os/toolkit/api_parser/proofpoint_trap/proofpoint_trap.py

```diff
@@ -0,0 +1,222 @@
+                            tmp_header["Authentication-Results"] = original_email['headers']["Authentication-Results"]
+
+                        if "Return-Path" in original_email['headers']:
+                            tmp_header["Return-Path"] = original_email['headers']["Return-Path"]
+
+                    original_email['headers'] = tmp_header
+
+                    # Drop useless parts
+                    original_email['body'] = ""
+                    original_email['mimeContent'] = ""
+                    original_email['urls'] = ""
+
+                    # Save the chosen email in new field 'email'
+                    alert['email'] = original_email
+
+                alert['emails'] = ""
+
+        return result
+
+    def test(self):
+
+        current_time = timezone.now()
+        try:
+            logs = self.get_logs(since=(current_time - timedelta(hours=12)), to=(current_time))
+
+            return {
+                "status": True,
+                "data": logs
+            }
+        except Exception as e:
+            logger.exception(f"{[__parser__]}:test: {str(e)}", extra={'frontend': str(self.frontend)})
+            return {
+                "status": False,
+                "error": str(e)
+            }
+
+    def get_logs(self, since=None, to=None):
+        alert_url = self.proofpoint_trap_host + "/api/incidents"
+
+        # Format timestamp for query
+        if isinstance(since, datetime):
+            since = since.isoformat()
+            since = since[:-13] + "Z"
+        # Format timestamp for query
+        if isinstance(to, datetime):
+            to = to.isoformat()
+            to = to[:-13] + "Z"
+
+        query = {}
+        # query['state'] = "open"
+
+        # logs with 'since' or 'to' values are included in return
+        if since:
+            query['created_after'] = since
+        if to:
+            query['created_before'] = to
+
+        return self.__execute_query(alert_url, query)
+
+    def format_logs(self, log):
+        return json.dumps(log)
+
+
+    def execute(self):
+
+        # self.frontend.last_api_call = timezone.now() - timedelta(days=30)
+
+
+        since = self.frontend.last_api_call or (timezone.now() - timedelta(days=30))
+
+        # fetch at most 24h of logs to avoid the process running for too long
+        to = min(timezone.now(), since + timedelta(hours=24))
+
+        msg = f"Parser starting from {since} to {to}"
+        logger.info(f"[{__parser__}]:execute: {msg}", extra={'frontend': str(self.frontend)})
+
+        response = self.get_logs(since, to)
+
+        # Downloading may take some while, so refresh token in Redis
+        self.update_lock()
+
+        logger.info(f"[{__parser__}]: GET LOG OK", extra={'frontend': str(self.frontend)})
+
+        for incendent_log in response:
+            alert_logs = incendent_log['events']
+            self.write_to_file([self.format_logs(log) for log in alert_logs])
```

Comment:
```suggestion
        for incident_log in response:
            alert_logs = incident_log['events']
            self.write_to_file([self.format_logs(log) for log in alert_logs])
```

did you mean "incident" ?